### PR TITLE
20221217 assign

### PR DESCRIPTION
**Pull request by prozacchiwawa**
this creates an "assign" form that performs destructuring and binding a lot like let, but with simpler and more versatile syntax.

    (mod (X)
      (include *standard-cl-21*)
      (defun-inline tup (X Y) (c X Y))
      (assign
        (X1 . X2) (tup (+ X 1) (+ X 2)) ;; 14
        X3 (+ X1 1) ;; 15
        X4 (+ X2 1) ;; 16
        (Y0 . X5) (tup (+ X4 1) (+ X4 1)) ;; 17
        X6 (+ Y0 1) ;; 18
        Y1 (+ X6 Y0) ;; 35
        Y1
        )
      )

Ordering is done automatically and the bindings are broken up into a stack of parallel let forms.

### CODE REVIEW

File: src/compiler/frontend.rs [hunk 1]
```diff
@@ -236,6 +238,145 @@ fn make_let_bindings(
...
+            LetData {
+                loc: l.clone(),
+                kw: Some(l.clone()),
+                bindings: binding_list,
+                body: Rc::new(output_let),
+            },
+        )
+    }
+
+    eprintln!("output let from assign {}", output_let.to_sexp());
```
Comment (trepca) [inline src/compiler/frontend.rs]: this should probably be removed

**Comment by prozacchiwawa** [comment]
Need to add additional tests in here, bug based on feedback.

**Comment by prozacchiwawa** [comment]
not actually a bug but adding tests surrounding this.
i'm going to push strict mode harder.

**Comment by prozacchiwawa** [comment]
https://github.com/Chia-Network/clvm_tools_rs/pull/124 <-- this bug was found, which is unrelated to this code, but is exercised by it.

**Review by AmineKhaldi — APPROVED** [review]

**Pull request merged**